File: src/main/java/com/baulsupp/oksocial/authenticator/oauth2/Oauth2Token.java
package com.baulsupp.oksocial.authenticator.oauth2;

import java.util.Optional;

public class Oauth2Token {
  public String accessToken;
  public Optional<String> refreshToken;
  public Optional<String> clientId;
  public Optional<String> clientSecret;

  public Oauth2Token(String accessToken) {
    this.accessToken = accessToken;
    this.refreshToken = Optional.empty();
  }

  public Oauth2Token(String accessToken, String refreshToken, String clientId, String clientSecret) {
    this.accessToken = accessToken;
    this.refreshToken = Optional.ofNullable(refreshToken);
    this.clientId = Optional.ofNullable(clientId);
    this.clientSecret = Optional.ofNullable(clientSecret);
  }
}

File: src/main/java/com/baulsupp/oksocial/authenticator/oauth2/Oauth2Token.kt
package com.baulsupp.oksocial.authenticator.oauth2

import java.util.Optional

class Oauth2Token {
    var accessToken: String
    var refreshToken: Optional<String>
    var clientId: Optional<String>
    var clientSecret: Optional<String>

    constructor(accessToken: String) {
        this.accessToken = accessToken
        this.refreshToken = Optional.empty()
    }

    constructor(accessToken: String, refreshToken: String, clientId: String,
                clientSecret: String) {
        this.accessToken = accessToken
        this.refreshToken = Optional.ofNullable(refreshToken)
        this.clientId = Optional.ofNullable(clientId)
        this.clientSecret = Optional.ofNullable(clientSecret)
    }
}
